src/Common.cpp

```diff
@@ -228,11 +228,13 @@ bool NAS2D::endsWith(const std::string& string, const std::string& end) noexcept
 
 bool NAS2D::startsWith(const std::string& string, char start)
 {
+	if (string.empty()) { return false; }
 	return *std::begin(string) == start;
```

Comment:
The logical operators `&&` and `||` are short-circuiting, which means they are safe to use for pre-condition checks. This could be rewritten as:
```cpp
return !string.empty() && (*std::begin(string) == start);
```

Reply:
Will do.